ducktape/cluster/remoteaccount.py

```diff
@@ -236,6 +236,35 @@ def monitor_log(self, log):
             offset = 0
         yield LogMonitor(self, log, offset)
 
+
+class iter_wrapper(object):
+    """
+    Helper class that wraps around an iterable object to provide has_next() in addition to next()
+    """
+    def __init__(self, iter_obj, descriptor=None):
+        self.iter_obj = iter_obj
+        self.descriptor = descriptor
+        self.sentinel = object()
+        self.cached = self.sentinel
+
+    def __iter__(self):
+        return self
+
+    def next(self):
+        if self.cached is self.sentinel:
+            return next(self.iter_obj)
+        next_obj = self.cached
+        self.cached = self.sentinel
+        return next_obj
+
+    def has_next(self, timeout_sec=None):
+        assert timeout_sec == None or self.descriptor != None, "should have descriptor to enforce timeout"
+
+        if self.cached is self.sentinel:
+            if timeout_sec == None or select.select([self.descriptor], [], [], timeout_sec)[0]:
```

Comment:
Ah.. thanks! I tested it on a standalone python file and copy past my changes to remoteaccount.py. But I forgot to add the import. Fixed now.